apiserver/crossmodel/crossmodel.go

```diff
@@ -239,7 +261,7 @@ func makeOfferParamsFromOffer(offer jujucrossmodel.ApplicationOffer) params.Appl
 // FindApplicationOffers gets details about remote applications that match given filter.
 func (api *API) FindApplicationOffers(filters params.OfferFilters) (params.FindApplicationOffersResults, error) {
 	var result params.FindApplicationOffersResults
-	offers, err := api.getApplicationOffersDetails(filters)
+	offers, err := api.getApplicationOffersDetails(filters, permission.ReadAccess)
```

Comment:
Add comments. Also, next PR will move Find and Show to a separate controller facade.